Editor/InstructionAnalyzers/GenericTypeInstantiationAnalyzer.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Unity.ProjectAuditor.Editor.CodeAnalysis;
+
+namespace Unity.ProjectAuditor.Editor.InstructionAnalyzers
+{
+    public class GenericTypeInstantiationAnalyzer : IInstructionAnalyzer
+    {
+        const int k_FirstDescriptorId = 500000;
+
+        Dictionary<string, ProblemDescriptor> m_GenericDescriptors = new Dictionary<string, ProblemDescriptor>();
```

Comment:
Nitpick really, but in other parts of the code this is called "Generics" with an s. Might be nice to name it consistently.